pkg/cluster/pod.go

```diff
@@ -283,13 +252,19 @@ func (c *Cluster) MigrateMasterPod(podName spec.NamespacedName) error {
 	}
 	// We may not have a cached statefulset if the initial cluster sync has aborted, revert to the spec in that case.
 	if *c.Statefulset.Spec.Replicas > 1 {
-		if masterCandidatePod, err = c.masterCandidate(oldMaster.Spec.NodeName); err != nil {
+		if masterCandidateName, err = c.getSwitchoverCandidate(oldMaster); err != nil {
 			return fmt.Errorf("could not find suitable replica pod as candidate for failover: %v", err)
 		}
 	} else {
 		c.logger.Warningf("migrating single pod cluster %q, this will cause downtime of the Postgres cluster until pod is back", c.clusterName())
 	}
 
+	masterCandidatePod, err := c.KubeClient.Pods(masterCandidateName.Namespace).Get(context.TODO(), masterCandidateName.Name, metav1.GetOptions{})
+
+	if err != nil {
+		return fmt.Errorf("could not get master candidate pod: %v", err)
+	}
+
```

Comment:
getSwitchoverCandidate only picks a replica when state is `running`. I don't think we need another API call here.

Reply:
Indeed but then I will need to change `getSwitchoverCandidate` to make it return a `Pod` instead of a `NamespacedName` as below methods require a Pod. Another alternative would be to create a new `getSwitchoverCandidatePod` and refactor  `getSwitchoverCandidate` to use that one if we want to keep the old signature. Which approach would you prefer ?

Reply:
ah I see. I think in this case, it's fine then. We would have to do the call anyway - whether it's here or in `getSwitchoverCandidate`. And the call does not happen each sync, so it's fine.